code/src/nuvla/api/api.py

```diff
@@ -75,25 +75,71 @@
 DEFAULT_COOKIE_FILE = os.path.expanduser('~/.nuvla/cookies.txt')
 HREF_SESSION_TMPL_PASSWORD = 'session-template/password'
 HREF_SESSION_TMPL_APIKEY = 'session-template/api-key'
+CLOUD_ENTRY_POINT_ID = 'cloud-entry-point'
 
 
 class NuvlaError(Exception):
-    def __init__(self, reason, response=None):
+    def __init__(self, reason, response: Optional[requests.Response] = None):
         super(NuvlaError, self).__init__(reason)
         self.reason = reason
         self.response = response
 
 
+class NuvlaResourceOperationNotAvailable(KeyError):
+    def __init__(self, reason, operation):
+        super().__init__(reason)
+        self.operation = operation
+
+
+def _request_debug(method, endpoint, params=None, doc=None,
+                   data=None, headers: Optional[dict]=None):
+    # TODO: Use logging library.
```

Comment:
@konstan There is a todo in the pull-request. Is it for later?

Reply:
@0xbase12 Yes, this is for later.

Reply:
@0xbase12 https://github.com/nuvla/python-library/issues/21